hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContract.php

```diff
@@ -193,21 +194,26 @@ protected static function getContractDates($contracts) {
     // If there are two (or more) Contracts starting on the same day
     // then we pick only the one with latest End date.
     foreach ($contracts['values'] as $contract) {
-      $details = civicrm_api3('HRJobDetails', 'getsingle', array(
+      $details = civicrm_api3('HRJobDetails', 'get', array(
```

Comment:
Agree, but as confirmed with Omar, currently we stick to API method as it provides current Details revision of requested Job Contract. We want to change it later but it would be more global change probably for every API method /  entity which use any additional logic.